mysql/resource_binlog.go

```diff
@@ -0,0 +1,129 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceBinLog() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateBinLog,
+		UpdateContext: UpdateBinLog,
+		ReadContext:   ReadBinLog,
+		DeleteContext: DeleteBinLog,
+		Importer: &schema.ResourceImporter{
+			StateContext: ImportDatabase,
+		},
+		Schema: map[string]*schema.Schema{
+			"retention_period": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NULL",
+			},
+		},
+	}
+}
+
+func CreateBinLog(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stmtSQL := binlogConfigSQL(d)
+	log.Println("Executing statement:", stmtSQL)
+
+	_, err = db.ExecContext(ctx, stmtSQL)
+	if err != nil {
+		return diag.Errorf("failed running SQL to set binlog retention period: %v", err)
+	}
+
+	d.SetId(d.Get("retention_period").(string))
+
+	return ReadBinLog(ctx, d, meta)
+}
+
+func UpdateBinLog(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stmtSQL := binlogConfigSQL(d)
+	log.Println("Executing statement:", stmtSQL)
+
+	_, err = db.ExecContext(ctx, stmtSQL)
+	if err != nil {
+		return diag.Errorf("failed updating binlog retention period: %v", err)
+	}
+
+	return ReadBinLog(ctx, d, meta)
+}
+
+func ReadBinLog(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	db, err := getDatabaseFromMeta(ctx, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	stmtSQL := "call mysql.rds_show_configuration"
+
+	log.Println("Executing query:", stmtSQL)
+	rows, err := db.QueryContext(ctx, stmtSQL)
+	if err != nil {
+		if mysqlErr, ok := err.(*mysql.MySQLError); ok {
+			if mysqlErr.Number == unknownDatabaseErrCode {
+				d.SetId("")
+				return nil
+			}
+		}
+		return diag.Errorf("Error verifying binlog retention period: %s", err)
+	}
+
+	results := make(map[string]string)
+	for rows.Next() {
+		var name, value, description string
```

Comment:
`value` is a nullable value--I think you'll run into errors in some cases. I think you can just use `sql.NullInt64` here (as the value) and `int64` as the value in the map.

Reply:
I end up using `var value sql.NullString` which pretty much has the same behavior.